Extract/Extract.ipynb

```diff
@@ -0,0 +1,264 @@
+   "source": [
+    "###Get position of annotations in text (Mat take a while for large files). Will be in downloads folder"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The total number of matches is 24\n"
+     ]
+    }
+   ],
+   "source": [
+    "'''\n",
+    "If total is off uncomment lines to check why\n",
+    "'''\n",
+    "root, ext = os.path.splitext(Path(file_path).name)\n",
+    "name = get_download_folder()\n",
+    "name = name + \"/\" + root + '.csv'\n",
+    "#TODO change 1522.csv to use identifier\n",
+    "#p = urllib.parse.urljoin(name, '1522.csv')\n",
+    "count = 0\n",
+    "with open(name, \"w\") as csv_file:\n",
+    "    writer = csv.writer(csv_file, delimiter=',')\n",
+    "    header = [\"Segment\", \"Start\", \"End\", \"Occurs\", \"Concept\"]\n",
+    "    writer.writerow(header)\n",
+    "    for index, row in df.iterrows():\n",
+    "        l = []\n",
+    "        seg = row['Segment']\n",
+    "        seg = re.sub(r'\\(', r'\\\\(', seg ) #Need to escape paraenthesis due to regex\n",
+    "        seg = re.sub(r'\\)', r'\\\\)', seg )\n",
+    "        seg = re.sub(r'\\?', r' ', seg )\n",
+    "        seg = re.sub(r'[^\\x00-\\x7F]+',' ', seg)\n",
+    "        seg = re.sub(r'\\s+', ' ', seg)\n",
+    "        seg = re.sub(r'\\\\r', ' ', seg)\n",
+    "        m = re.search(seg, text, re.M | re.DOTALL)\n",
+    "        url =  'http://diging.asu.edu/uri/leuphana' + str(row['Dokumentname'])\n",
+    "        #print(seg)\n",
+    "        if m:\n",
+    "            found = m.group(0)\n",
+    "            l.append(str(found))\n",
+    "            l.append(str(m.start()))\n",
+    "            l.append(str(m.end()))\n",
+    "            l.append(url)\n",
+    "            l.append( row['Code'])\n",
+    "            writer.writerow(l)\n",
+    "            #print(m.start(), m.end())\n",
+    "            #print(\"FOUND\")\n",
+    "            #print(found)\n",
+    "            count += 1\n",
+    "        else:\n",
+    "            seg = re.sub(r'\\W+', '\\\\W+', seg)\n",
+    "            m = re.search(seg, text, re.M | re.DOTALL)\n",
+    "            if m:\n",
+    "                pattern = re.compile(r'1522.txt')\n",
+    "                found = m.group(0)\n",
+    "                l.append(str(found))\n",
+    "                l.append(str(m.start()))\n",
+    "                l.append(str(m.end()))\n",
+    "                l.append(url)\n",
+    "                l.append( row['Code'])\n",
+    "                writer.writerow(l)\n",
+    "                #print(m.start(), m.end())\n",
+    "                #print(\"FOUND\")\n",
+    "                #print(found)\n",
+    "                count += 1\n",
+    "\n",
+    "            \n",
+    "        #print(\"***********************\")\n",
+    "print(\"The total number of matches is {}\".format(count))"
```

Comment:
remove all commented lines that are not needed anymore

Reply:
removed